### PR TITLE
[11.0][ADD] filter_multi_user

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Allows to share user-defined filters filters among several users.

@ForgeFlow

### CODE REVIEW

File: filter_multi_user/models/ir_filters.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    _inherit = "ir.filters"
+
+    user_ids = fields.Many2many(
+        comodel_name="res.users",
+        string="Users",
+    )
+
+    @api.model
+    def get_filters(self, model, action_id=None):
+        """WARNING: this function overrides the standard one.
```
Comment (simahawk) [inline filter_multi_user/models/ir_filters.py]: pls do not override the docstring. Use a comment instead.

**Review by simahawk — APPROVED** [review]
LG, small remark on docstring.

File: filter_multi_user/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "name": "Filter Multi User",
+    "summary": "Allows to share user-defined filters filters among several users.",
+    "version": "11.0.1.0.0",
+    "development_status": "Beta",
+    "maintainers": ['LoisRForgeFlow'],
+    "category": "Tools",
+    "website": "https://github.com/OCA/server-ux",
+    "author": "ForgeFlow, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
```
Comment (etobella) [inline filter_multi_user/__manifest__.py]: Nitpicking: Not necessary

**Review by etobella — APPROVED** [review]
LGTM Just one comment

**Review by jarroyomorales — CHANGES_REQUESTED** [review]
LGTM! Functional and code review, but could you add some tests? Thank you!

File: filter_multi_user/README.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+To share a filter among several users you need to:
+
+#. Go to *Settings > Technical > User-defined Filters*.
```
Comment (HaraldPanten) [inline filter_multi_user/README.rst]:
```suggestion
#. Go to *Settings > Technical > User Interface > User-defined Filters*.
```

**Review by HaraldPanten — APPROVED** [review]
LGTM 👍

File: filter_multi_user/models/ir_filters.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply filter_multi_user/models/ir_filters.py]: Fixed :)

File: filter_multi_user/__manifest__.py (hunk 2 above)
Comment (LoisRForgeFlow) [reply filter_multi_user/__manifest__.py]: Fixed :)

File: filter_multi_user/README.rst (hunk 3 above)
Comment (LoisRForgeFlow) [reply filter_multi_user/README.rst]: Fixed :)

**Comment by LoisRForgeFlow** [comment]
@jarroyomorales Tests added, could you update your review? :D

**Review by jarroyomorales — APPROVED** [review]

File: filter_multi_user/tests/test_filter_multi_user.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+from odoo.tests import common
+from odoo.exceptions import AccessError
+
+
+class TestFilterMultiUser(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.filter_model = cls.env["ir.filters"]
```
Comment (simahawk) [inline filter_multi_user/tests/test_filter_multi_user.py]:
```suggestion
        super().setUpClass()
        cls.env = cls.env(
            context=dict(cls.env.context, tracking_disable=True, no_reset_password=True)
        )
        cls.filter_model = cls.env["ir.filters"]
```

File: filter_multi_user/tests/test_filter_multi_user.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+        cls.group_user = cls.env.ref("base.group_user")
+
+        cls.user_1 = cls._create_user("user_1", [cls.group_user])
+        cls.user_2 = cls._create_user("user_2", [cls.group_user])
+
+    @classmethod
+    def _create_user(self, login, groups):
+        group_ids = [group.id for group in groups]
+        user = self.user_model.with_context({"no_reset_password": True}).create({
```
Comment (simahawk) [inline filter_multi_user/tests/test_filter_multi_user.py]:
```suggestion
        user = self.user_model.create({
```

**Comment by LoisRForgeFlow** [comment]
Well, let's move on with this one. Thanks for reviews!

/ocabot merge nobump

**Pull request merged**